node/src/prover/mod.rs

```diff
@@ -119,6 +117,10 @@ impl<N: Network, C: ConsensusStorage<N>> Prover<N, C> {
             matches!(storage_mode, StorageMode::Development(_)),
         )
         .await?;
+
+        // Initialize the sync module.
+        let sync = BlockSync::new(BlockSyncMode::Router, ledger_service.clone(), router.tcp().clone());
```

Comment:
@raychu86 can you review to confirm this reordering is safe? IIRC this has initialization implications to the node's state.